### DISCUSSION TITLE
[Feature]: Support import.meta.url

### DISCUSSION THREAD

**Discussion by ksmithut** [remix-run/remix#2607] [Proposals]
### What is the new or updated feature that you are suggesting?

Add support for `import.meta.url` as an alternative to `__filename` and `__dirname`

### Why should this feature be included?

It is a JavaScript/Web standard for getting the url to the current file. It's supported in node, deno, as well as the web. The alternatives `__dirname` and `__filename` are supported in the CommonJS variant of node, but not when using ECMAScript Modules. You can easily recreate these variables with `const __dirname = new URL('./', import.meta.url).pathname`. #useThePlatform

**Comment by candidia** [comment]
I believe the lack of this support is what causes dicebear to not work with Remix https://github.com/dicebear/dicebear/issues/321

**Comment by andrecasal** [comment]
I'm building a UI library for Remix/React and want the component's source code that's shown on the website to be the one from the components code on disk. This is the current state of afairs, in case you need the same.

```tsx
import { promises as fs } from 'fs'
import { fileURLToPath } from 'url'
import path, { dirname } from 'path'
import { useLoaderData } from '@remix-run/react'

export const loader = async () => {
	const basePath = '../app/routes/'
	const __filename = fileURLToPath(import.meta.url)
	const __dirname = dirname(__filename)
	const filePath = path.join(__dirname, basePath, '/_marketing+/ui+/components/typography/heading.tsx')
	const source = await fs.readFile(filePath, 'utf-8')
	return json({ source })
}
```
